File: src/com/OpenSerenity/elements/NativeElement.java
package com.OpenSerenity.elements;

import org.openqa.selenium.WebElement;

import java.util.Collection;
import java.util.List;

public interface NativeElement {
    String getTagName() throws Exception;
    String getText() throws Exception;
    boolean isEnabled() throws Exception;
    boolean isSelected() throws Exception;
    boolean isDisplayed() throws Exception;
    boolean isExists();
    void clear() throws Exception;
    void sendKeys(String text) throws Exception;
    void submit() throws Exception;
    void click() throws Exception;
    void altClick();
    String getAttribute(String attributeName) throws Exception;
    String getCssValue(String propertyName) throws Exception;
    NativeElement findChild(String locator);
    List<NativeElement> findChildren(String locator);
}

File: src/scala/com/OpenSerenity/elements/NativeElement.scala
package com.OpenSerenity.elements

import scala.Predef.String

trait NativeElement {
  def getTagName(): String
  def getText(): String
  def isEnabled(): Boolean
  def isSelected(): Boolean
  def isDisplayed(): Boolean
  def isExists(): Boolean
  def clear()
  def sendKeys(text: String)
  def submit()
  def click()
  def altClick()
  def getAttribute(attributeName: String): String
  def getCssValue(propertyName: String): String
  def findChild(locator: String): NativeElement
  def findChildren(locator: String): List[NativeElement]
}


